### ISSUE TITLE
Auto capitlising URL still occurring

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#1440]
## Describe the bug

When enter a URL during initialization, the `h` in `http` gets auto-capitlized. The fix here was supposed to stop it: #1351 but on some devices it still seems to occur

### To reproduce

1. Go to initialize
2. Enter a URL
3. `H` capitilized

### Expected behaviour

Should not be so confusing for users and should just lower case `http`? @wlthomson ?

### Version and device info

Not sure, @alainsussol can you let us know the specs of your tablet please? 

### Additional context

N/A

**Comment by wlthomson** [comment]
Hmm, this is strange. The fix is declarative, no logic that could be going wrong (just passing prop `autocorrect={false}`), and tested working on emulator and tablet. Can we replicate this on any of our tablets (and are we 100% sure the replication is being done on a tablet with the latest develop?).

Possible related bug: https://github.com/facebook/react-native/issues/18457.

**Comment by alainsussol** [comment]
@joshxg @wlthomson 

My tablet has the following spec

**DISPLAY**
Display Type						**TFT**
Screen Size						**10.1 in**
Native Screen Resolution			**1920 x 1200**
Touchscreen						**Yes**
3D Capable Display					**No**
Screen Finish						**Glossy**

**STORAGE**
Built-in Storage - Capacity GB		**32 GB**
Built-in Storage Type				**Flash Memory**
Built-in Memory Card Reader			**Yes**
Compatible Memory Card Types		**microSD**
Max Memory Card Capacity			**512 GB**
RAM Size							**2 GB**

**CAMERAS**
Rear Camera Still Resolution		        **8 MP**
Rear Camera Video Resolution		**1920 x 1080 (FHD) @ 30fps**
Rear Camera Autofocus				**Yes**
Rear Camera Flash					**No**
Front Camera Still Resolution		        **5 MP**
Front Camera Autofocus				**No**
Front Camera Flash					**No**
Video Calling (Wi-Fi)			        **Yes**
Video Calling (Cellular)			        **Yes**
3D Photos						**No**
3D Video Recording				**No**

**PROCESSOR**
Processor Speed					**1.6 GHz**
Processor Cores					**8**

**SOFTWARE**
Operating System					**Android 9.0**
Operating System Language			**Bilingual**
Multitasking						**Yes**
App Store Compatibility				**Google Play Store**

**AUDIO**
Built-in Speaker					**Yes**
Integrated Microphone				**Yes**
Microphone Input					**No**
Hardware Volume Control			**Yes**
Headphone Jack					**Yes**

**CONNECTIVITY**
Integrated Wi-Fi					**802.11 a/b/g/n/ac 2.4G+5GHz, VHT80**
Integrated Bluetooth				**Yes - Bluetooth 5.0**
3G								**No**
4G								**No**
4G LTE							**No**
Assisted GPS Navigation				**No**
Unassisted GPS Navigation			**Yes**
HDMI Output						**No**
DLNA Certified					**Yes**

**POWER**
Approximate In-use Battery Life		**13 h**
Battery - Capacity					**6150 mAh**
Charge over USB					**Yes**

**PHYSICAL FEATURES**
Sensors                                                **ACCELEROMETER**			
Hard-key QWERTY Keyboard			**No**
Stylus							**No**
Built-in Stand						**No**
Colour							**Black**
Height							**25.62 cm**
Depth							**3.64 cm**
Width							**16.09 cm**
Height (Inches)					**10.09 in**
Depth (Inches)						**1.43 in**
Width (Inches)						**6.33 in**
Weight							**469 g**

**WARRANTY**
Manufacturer's Warranty - Parts		**1 Year**
Manufacturer's Warranty - Labour	        **1 Year**

**Comment by alainsussol** [comment]
@joshxg @wlthomson 

I am testing with develop 406RC3

**Comment by wlthomson** [comment]
> @joshxg @wlthomson
> 
> I am testing with develop 406RC3

Sorry, I was referring to the mobile version (this functionality is independent of any desktop stuff). Are you sure you're testing with an up-to-date .apk?

**Comment by alainsussol** [comment]
@wlthomson V3.0.0

**Comment by Chris-Petty** [comment]
Is it possible it isn't working cause it's the wrong prop? https://facebook.github.io/react-native/docs/textinput#autocapitalize

**Comment by diegoevangelisti** [comment]
What if, even if the `h` is being autocapitalised, we convert it to lower case before initialising the synchroniser? 

https://github.com/openmsupply/mobile/blob/5ee0701159e9267901a7ccc4e7e76a97484d2404/src/pages/FirstUsePage.js#L41
(that simple line of code can be improved, open to suggestions)

That way we can assure an `h` during initialisation.

**Comment by joshxg** [comment]
The prop added in this PR: https://github.com/openmsupply/mobile/pull/1351/files was the wrong prop, so as chris says it needs to be changed to https://facebook.github.io/react-native/docs/textinput#autocapitalize

I like your suggestion - it will also need to be changed in `SettingsPage`, where you can change the server url. I think for this issue, just change the prop. We can make the url lower case in this issue: https://github.com/openmsupply/mobile/issues/1539

**Comment by wlthomson** [comment]
> What if, even if the `h` is being autocapitalised, we convert it to lower case before initialising the synchroniser?
> 
> https://github.com/openmsupply/mobile/blob/5ee0701159e9267901a7ccc4e7e76a97484d2404/src/pages/FirstUsePage.js#L41
> 
> (that simple line of code can be improved, open to suggestions)
> 
> That way we can assure an `h` during initialisation.

I thought of this initially but I didn't go with this is because in general I think it's a bit confusing for the user if the capitalisation is done behind the scenes rather than using the props which are tried and tested and behave in a way the user is familiar with, as long as you use the right one :). 
The situation to be avoided (which is unlikely here) would be where a user purposefully capitalises something, and behind the scenes the code runs `.toLowerCase()` without any UX feedback.

TLDR; I think this is the use-case for `autoCapitalise={"none"}`, which should do the job nicely :). Might also be worth leaving `autoCorrect={false}`, as it could also be frustrating to have annoying corrections when entering long URLs.

**Comment by diegoevangelisti** [comment]
Thanks @joshxg  and @wlthomson for your feedback and suggestions. No worries about the length of your explanations, more than welcome here. I will proceed with `autoCapitalise={"none"}` and `autoCorrect={false}` and create the PR if test goes well.

**Comment by joshxg** [comment]
Closed by #1552

**Comment by joshxg** [comment]
autoCapitalize={"none"}
- [x] Go to initialize, type a server URL which starts with `h`, hit Enter and the `h` doesn't change to `H` (e.g. `http://10.2.0.0:3000`, results in: `http://10.2.0.0:3000`)
